compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -1881,6 +1881,16 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       end extension
     end MatchTypeMethods
 
+    type MatchTypeCase = dotc.core.Types.Type
+
+    object MatchTypeCase extends MatchTypeCaseModule:
+      def unapply(x: MatchTypeCase): Option[(List[String], List[TypeBounds], TypeRepr, TypeRepr)] =
```

Comment:
We can implement this in userspace, this should not be added here. It is unclear what the ideal API should be, it would be better to implement such custom extractors in a separate library.